test/Fencer/Rules/Test.hs

```diff
@@ -26,20 +28,85 @@ tests = testGroup "Rule tests"
   [ test_rulesLoadRulesYaml
   , test_rulesLoadRulesNonYaml
   , test_rulesLoadRulesRecursively
+  , test_rulesLoadRulesDotDirectory
+  , test_rulesLoadRulesRUNTIME_IGNOREDOTFILEStrue
+  , test_rulesLoadRulesRUNTIME_IGNOREDOTFILESfalse
   ]
 
+-- | Create given directory structure and check that 'loadRulesFromDirectory'
+-- produces expected result.
+expectLoadRules
+  :: "ignoreDotFiles" :! Bool
+  -> "files" :! [(FilePath, Text)]
+  -> "result" :! [DomainDefinition]
+  -> Assertion
+expectLoadRules
+  (arg #ignoreDotFiles -> ignoreDotFiles)
+  (arg #files -> files)
+  (arg #result -> result) =
+  Temp.withSystemTempDirectory "fencer-config" $ \tempDir -> do
+    forM_ files $ \(path, txt) -> do
+      let (dir, file) = splitFileName path
+      createDirectoryIfMissing True (tempDir </> dir)
+      TIO.writeFile (tempDir </> dir </> file) txt
+    definitions <- loadRulesFromDirectory
+      (#rootDirectory tempDir)
+      (#subDirectory ".")
+      (#ignoreDotFiles ignoreDotFiles)
+    assertEqual "unexpected definitions"
+      (sortOn domainDefinitionId result)
+      (sortOn domainDefinitionId definitions)
+
 -- | test that 'loadRulesFromDirectory' loads rules from YAML files.
 test_rulesLoadRulesYaml :: TestTree
 test_rulesLoadRulesYaml =
   testCase "Rules are loaded from YAML files" $
-    Temp.withSystemTempDirectory "fencer-config" $ \tempDir -> do
-      TIO.writeFile (tempDir </> "config1.yml") domain1Text
-      TIO.writeFile (tempDir </> "config2.yaml") domain2Text
-      definitions <-
-        loadRulesFromDirectory (#directory tempDir) (#ignoreDotFiles True)
-      assertEqual "unexpected definitions"
-        (sortOn domainDefinitionId [domain1, domain2])
-        (sortOn domainDefinitionId definitions)
+    expectLoadRules
+      (#ignoreDotFiles True)
+      (#files
+        [ ("config1.yml", domain1Text)
+        , ("config2.yaml", domain2Text) ]
+      )
+      (#result [domain1, domain2])
+
+-- | test that 'loadRulesFromDirectory' does not load rules from a
+-- dot-directory when dot-files should be ignored.
+test_rulesLoadRulesDotDirectory :: TestTree
+test_rulesLoadRulesDotDirectory =
+  testCase "Rules are not loaded from a dot-directory" $
+    expectLoadRules
+      (#ignoreDotFiles True)
+      (#files
+        [ (".domain1" </> "config1.yml", domain1Text)
+        , ("domain2" </> "config2.yaml", domain2Text) ]
+      )
+      (#result [domain2])
+
+-- | test that 'loadRulesFromDirectory' correctly implements the case
+-- RUNTIME_IGNOREDOTFILES=true.
+test_rulesLoadRulesRUNTIME_IGNOREDOTFILEStrue :: TestTree
```

Comment:
Can you rename it to `test_rulesLoadRules_ignoreDotFiles` or something? It is not related to the env var anymore, it's a unit test for `loadRulesFromDirectory` basically.

Reply:
Sure! Done in an upcoming commit.